### DISCUSSION TITLE
Need help of checking why env is not set on windows

### DISCUSSION THREAD

**Discussion by prisis** [community/community#25593] [Actions]
<p>Was trying many stuff out and it was allways null, so this is what i get as a output</p>
<pre><code>$COMPONENTS.Split(",") | ForEach {
    echo ::group::$_
    cd $_
    composer validate --strict
    $COMPOSER_UP
    $PHPUNIT
    cd $GITHUB_WORKSPACE
    echo ::endgroup::
  }
  shell: C:\windows\System32\WindowsPowerShell\v1.0\powershell.EXE -command "&amp; '{0}'"
  env:
    PHPUNIT: ./vendor/bin/phpunit --verbose
    COMPOSER_UP: composer update --no-progress --no-suggest --no-interaction --prefer-source --ansi
    COMPONENTS: src/Viserio/Bridge/Dotenv,src/Viserio/Bridge/Monolog,src/Viserio/Bridge/Phpstan,src/Viserio/Bridge/Twig,src/Viserio/Component/Bus,src/Viserio/Component/Cache,src/Viserio/Component/Config,src/Viserio/Component/Console,src/Viserio/Component/Container,src/Viserio/Component/Cookie,src/Viserio/Component/Cron,src/Viserio/Component/Events,src/Viserio/Component/Exception,src/Viserio/Component/Filesystem,src/Viserio/Component/Foundation,src/Viserio/Component/Http,src/Viserio/Component/HttpFactory,src/Viserio/Component/HttpFoundation,src/Viserio/Component/Log,src/Viserio/Component/Mail,src/Viserio/Component/Manager,src/Viserio/Component/OptionsResolver,src/Viserio/Component/Pagination,src/Viserio/Component/Parser,src/Viserio/Component/Pipeline,src/Viserio/Component/Profiler,src/Viserio/Component/Routing,src/Viserio/Component/Session,src/Viserio/Component/Support,src/Viserio/Component/Translation,src/Viserio/Component/Validation,src/Viserio/Component/View,src/Viserio/Component/WebServer,src/Viserio/Contract,src/Viserio/Provider/Debug,src/Viserio/Provider/Twig,
You cannot call a method on a null-valued expression.
At D:\a\_temp\b55bfeae-b30e-47b9-bd7c-bf4260f9a396.ps1:3 char:1
+ $COMPONENTS = $env.COMPONENTS.TrimEnd(",")
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo : InvalidOperation: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvokeMethodOnNull

##[error]Process completed with exit code 1.
</code></pre>
<p>I tryed $env.COMPONENTS and $COMPONENTS but it is always null, but in the env output its there.</p>
<p>Link to the file <a href="https://github.com/narrowspark/framework/blob/ecd4759b173f35d0c177ce703eb1ae09ff41f29f/.github/workflows/continuous-integration.yml#L176" rel="nofollow noopener">https://github.com/narrowspark/framework/blob/ecd4759b173f35d0c177ce703eb1ae09ff41f29f/.github/workflows/continuous-integration.yml#L176</a> hope someone has a idea</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>I find your another thread with the same question: <a href="https://github.community/t/env-is-null-on-windows-operating-system/16867">https://github.community/t5/GitHub-Actions/Env-is-null-on-windows-operating-system/m-p/37979/highlight/false#M3174</a> , I have replied on that one.  </p>
<p>Please follow it for any updates. </p>
